### PR TITLE
Drop NetTrail - emit iterations metrics in runner instead of dialer

### LINKED ISSUE #1605: setup() function creates an iteration metric
`setup()` creates an iteration metric.

## Environment

- k6 version: 0.27.0
- OS and version: Linux / cloud execution

## Expected Behavior

I expected to see a single iteration.

## Actual Behavior
`setup()` function created a separate iteration. Interestingly the `teardown()` function doesn't create a new iteration.

## script
```javascript
import http from 'k6/http'
import { sleep } from 'k6';

export let options = {
  iterations: 1,
  vus: 1
}

export function setup() { // creates 1 iteration
  let res = http.get('http://test.k6.io?where=setup')
}
export default function () { // created 1 iteration
  sleep(10);
  let res = http.get('http://test.k6.io?where=defaultfunction')
}

export function teardown() { // created 0 iterations
  let res = http.get('http://test.k6.io?where=teardown')
}
```

Cloud link: https://app.k6.io/runs/public/7c0d02656de748608dfd68eed2f701c2

image

Unrelated side note: It's interesting that the total count of requests is 2. The request from the `teardown()` function isn't counted.

### PR DESCRIPTION
**Pull request by mstoykov**
## What?

Drop the NetTrail container and emit iterations metric in runner instead of in Dialer.

Also fix #1605 

## Why?

As part of looking over issues I found #1250 which has some additional work required. But dropping the specific NetTrail and moving the code around was really simple. 

FIxing some of the test took a bit more time :(

## Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have added tests for my changes.
- [ ] I have run linter locally (`make lint`) and all checks pass.
- [ ] I have run tests locally (`make tests`) and all tests pass.
- [ ] I have commented on my code, particularly in hard-to-understand areas.

## Related PR(s)/Issue(s)
#1250 
#1605

### CODE REVIEW

File: execution/scheduler_ext_test.go [hunk 1]
```diff
@@ -497,10 +501,12 @@ func TestSchedulerRunCustomTags(t *testing.T) {
 							gotTrailTag = true
 						}
 					}
-					if netTrail, ok := sample.(*netext.NetTrail); ok && !gotNetTrailTag {
-						tags := netTrail.Tags.Map()
-						if v, ok := tags["customTag"]; ok && v == "value" {
-							gotNetTrailTag = true
+					for _, s := range sample.GetSamples() {
+						if s.Metric.Name == metrics.DataSentName && !gotNetTrailTag {
```
Comment (olegbespalov) [inline execution/scheduler_ext_test.go:505]: nit: after the change, this `gotNetTrailTag` could confuse

File: lib/netext/dialer.go [hunk 2]
```diff
@@ -69,17 +69,14 @@ func (d *Dialer) DialContext(ctx context.Context, proto, addr string) (net.Conn,
 	return conn, err
 }
 
-// GetTrail creates a new NetTrail instance with the Dialer
+// Sample creates a new NetTrail instance with the Dialer
```
Comment (olegbespalov) [inline lib/netext/dialer.go:72]: How about naming this something like `ReadsWritesSamples` or `IOSamples`, something more precise? Also, a method comment not longer valid :cry:

**Review by olegbespalov — COMMENTED** [review]
Looks good, left two minors

File: lib/netext/dialer.go (hunk 2 above)
Comment (mstoykov) [reply lib/netext/dialer.go]: I meant it more like hte verb "sample" as in dialer samples whatever it has to sample and returns it. But it isn't a thing we usually do so I had it renamed

**Review by olegbespalov — APPROVED** [review]

**Review by joanlopez — APPROVED** [review]

**Pull request merged**